### PR TITLE
Live updates

### PR DESCRIPTION
**Pull request by justcalamari**
Implemented automatic updates.

### CODE REVIEW

**Comment by justcalamari** [comment]
@sambrv @EthanSchoen Ready for review.

File: src/codeu/chat/client/ClientContext.java [hunk 1]
```diff
@@ -19,16 +19,44 @@
...
-    user = new ClientUser(controller, view);
-    conversation = new ClientConversation(controller, view, user);
-    message = new ClientMessage(controller, view, user, conversation);
-  }
+    private static final int SERVER_REFRESH_MS = 5000;  // 5 seconds
+
+    public final ClientUser user;
+    public final ClientConversation conversation;
+    public final ClientMessage message;
+
```
Comment (narendergupta) [inline src/codeu/chat/client/ClientContext.java]: Extra newline.

File: src/codeu/chat/client/ClientContext.java [hunk 2]
```diff
@@ -19,16 +19,44 @@
...
 import codeu.chat.client.View;
+import codeu.chat.util.Timeline;
+import codeu.chat.util.Logger;
 
 public final class ClientContext {
 
-  public final ClientUser user;
-  public final ClientConversation conversation;
-  public final ClientMessage message;
+    private static final Logger.Log LOG = Logger.newLog(ClientContext.class);
```
Comment (narendergupta) [inline src/codeu/chat/client/ClientContext.java]: Extra newline.

File: src/codeu/chat/client/ClientContext.java [hunk 3]
```diff
@@ -19,16 +19,44 @@
...
 
-  public ClientContext(Controller controller, View view) {
-    user = new ClientUser(controller, view);
-    conversation = new ClientConversation(controller, view, user);
-    message = new ClientMessage(controller, view, user, conversation);
-  }
+    private static final int SERVER_REFRESH_MS = 5000;  // 5 seconds
```
Comment (narendergupta) [inline src/codeu/chat/client/ClientContext.java]: Some files have 2-space tab while others have 4-space tabs. Please make them consistent.

File: src/codeu/chat/client/ClientContext.java [hunk 4]
```diff
@@ -19,16 +19,44 @@
...
+    public ClientContext(Controller controller, View view) {
+        user = new ClientUser(controller, view);
+        conversation = new ClientConversation(controller, view, user);
+        message = new ClientMessage(controller, view, user, conversation);
+
+        timeline.scheduleNow(new Runnable() {
+            @Override
+            public void run() {
+                try {
+
```
Comment (narendergupta) [inline src/codeu/chat/client/ClientContext.java]: Lots of extra newlines. Please fix.

File: src/codeu/chat/client/ClientContext.java [hunk 5]
```diff
@@ -19,16 +19,44 @@
...
+                    LOG.info("Reading from server...");
+
+                    //conversation.updateAllConversations(false);
```
Comment (narendergupta) [inline src/codeu/chat/client/ClientContext.java]: Add a comment for why the line is commented out, or remove it altogether.

File: src/codeu/chat/client/simplegui/ChatSimpleGui.java [hunk 6]
```diff
@@ -115,5 +120,26 @@ private void initialize() {
 
     mainFrame.add(mainViewPanel);
     mainFrame.pack();
+
+    timeline.scheduleNow(new Runnable() {
+        @Override
```
Comment (narendergupta) [inline src/codeu/chat/client/simplegui/ChatSimpleGui.java]: Inconsistent 2-4 spaced tabs.

**Review by narendergupta — CHANGES_REQUESTED** [review]

**Review by EthanSchoen — APPROVED** [review]

**Pull request merged**